### ISSUE TITLE
HA masters sometimes assign the same nodeport to multiple services

### ISSUE THREAD

**Issue by carsonoid** [kubernetes/kubernetes#44098]
**BUG REPORT**

**Kubernetes version** (use `kubectl version`): 1.5.1

**Environment**:
- **Cloud provider or hardware configuration**: AWS
- **OS** (e.g. from /etc/os-release): Ubuntu 16.04
- **Kernel** (e.g. `uname -a`):  4.4.0-66-generic 
- **Install tools**: salt
- **Others**:

**What happened**:

When running multiple API servers behind a loadbalancer it is possible to get the same nodeport assigned to multiple services. We have seen a nodeport re-used up to 4 times. 

When a nodeport gets re-used it prevents the kube-proxy iptables from being created: https://github.com/kubernetes/kubernetes/blob/v1.5.2/pkg/proxy/iptables/proxier.go#L1108 which means all services that get a duplicated nodeport after the first one fail on all nodes.

**What you expected to happen**:

Nodeports assigned via the API should always be unique across the cluster.

**How to reproduce it** (as minimally and precisely as possible):

 * Run multiple apiservers behind a loadbalancer (we have 3)
 * Create many nodeport enabled services at the same time. (we used gnu parallel to do this)
 * Check the created services and wait for duplicates to show up

We Created 1200 services, 5 at a time concurrently, each with 2 NodePorts. Ended up with 72 or 6% of those NodePorts conflicting with another.

**Comment by coryfklein** [comment]
To reproduce this, I created 1200 json files representing services:

```
#!/bin/bash

NAMESPACE=wee-personal2
END=1200
SERVICE='{ "apiVersion": "v1", "kind": "Service", "metadata": { "name": "NAME_REPLACE", "namespace": "NAMESPACE_REPLACE" }, "spec": { "ports": [ { "name": "primary", "port": 80, "protocol": "TCP", "targetPort": "primary" }, { "name": "exporter", "port": 9104, "protocol": "TCP", "targetPort": "exporter" } ], "selector": { "app": "mysql" }, "sessionAffinity": "None", "type": "NodePort" } }'

mkdir $NAMESPACE

for i in $(seq 1 $END); do
    echo $SERVICE | sed "s/NAME_REPLACE/cory-${i}/" | sed "s/NAMESPACE_REPLACE/${NAMESPACE}/" > ${NAMESPACE}/cory-${i}.json
done
```

Then created the services 5 at a time:

```
cd wee-personal2
ls | parallel -n 1 -P 5 kubectl create -f
```

Then count the number of services that have conflicting NodePorts:

    kubectl get service -o yaml --all-namespaces | grep "nodePort" | sed -e 's/^[[:space:]]*nodePort: //' | sort -d | uniq -c

**Comment by carsonoid** [comment]
I think this may have been fixed in #43259
